### PR TITLE
Support symlink for NioFSWrappingFS

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt [hunk 1]
```diff
@@ -36,24 +36,29 @@ import okio.Path.Companion.toOkioPath
...
-  // TODO(Benoit) How do deal with multiple directories? Test with Windows someday.
-  private val delegateRoot = javaNioFileSystem.rootDirectories.first()
-
+internal class NioFileSystemWrappingFileSystem(private val nioFileSystem: NioFileSystem) : NioSystemFileSystem() {
   /**
    * On a `java.nio.file.FileSystem`, `java.nio.file.Path` are stateful and hold a reference to the file system they
    * got provided from. We need to [resolve][java.nio.file.Path.resolve] all okio paths before doing operations on the
    * nio file system in order for things to work properly.
    */
   private fun Path.resolve(readSymlink: Boolean = false): NioPath {
```
Comment (oldergod) [inline okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt:45]: Maybe I could rename the method to something else

**Review by oldergod — COMMENTED** [review]
I got confused a lot by JimFs (and its FS being in memory) but I think it's good now!

File: okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt [hunk 2]
```diff
@@ -74,17 +79,7 @@ internal class NioFileSystemWrappingFileSystem(javaNioFileSystem: JavaNioFileSys
...
-      val path = dir / entry.toOkioPath()
-      if (dir.isRelative) {
-        // All `entries` are absolute and resolving them against `dir` won't have any effect. We need to manually
-        // relativize them back.
-        nioDir.relativize(path.resolve()).toOkioPath()
-      } else {
-        path
-      }
-    }
+    val result = entries.mapTo(mutableListOf()) { entry -> dir / entry.toString() }
```
Comment (oldergod) [inline okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt:82]: everything's simpler once you start using it correctly....

File: okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt [hunk 3]
```diff
@@ -195,7 +190,14 @@ internal class NioFileSystemWrappingFileSystem(javaNioFileSystem: JavaNioFileSys
   }
 
   override fun createSymlink(source: Path, target: Path) {
-    Files.createSymbolicLink(source.resolve(), target.resolve())
+    val sourceNioPath = source.resolve()
+    val targetNioPath =
+      if (source.isAbsolute && target.isRelative) {
```
Comment (swankjesse) [inline okio/src/jvmMain/kotlin/okio/NioFileSystemWrappingFileSystem.kt:195]: Why is this necessary? Relative symlinks should be okay?

**Review by swankjesse — APPROVED** [review]

File: okio-testing-support/src/commonMain/kotlin/okio/AbstractFileSystemTest.kt [hunk 4]
```diff
@@ -2211,6 +2247,7 @@ abstract class AbstractFileSystemTest(
   @Test
   fun openSymlinkSink() {
     if (!supportsSymlink()) return
+    if (isJimFileSystem()) return
```
Comment (oldergod) [inline okio-testing-support/src/commonMain/kotlin/okio/AbstractFileSystemTest.kt:2250]: https://github.com/google/jimfs/issues/254
